python/felis/cli.py

```diff
@@ -117,48 +117,56 @@ def create(
-        schema.name = schema_name
-    elif url.drivername == "sqlite":
-        logger.info("Overriding schema name for sqlite with: main")
-        schema.name = "main"
-    if not url.host and not url.drivername == "sqlite":
-        dry_run = True
-        logger.info("Forcing dry run for non-sqlite engine URL with no host")
-
-    metadata = MetaDataBuilder(schema).build()
-    logger.debug(f"Created metadata with schema name: {metadata.schema}")
-
-    engine: Engine | MockConnection
-    if not dry_run and not output_file:
-        engine = create_engine(url, echo=echo)
-    else:
-        if dry_run:
-            logger.info("Dry run will be executed")
-        engine = DatabaseContext.create_mock_engine(url, output_file)
-        if output_file:
-            logger.info("Writing SQL output to: " + output_file.name)
-
-    context = DatabaseContext(metadata, engine)
-
-    if drop_if_exists:
-        logger.debug("Dropping schema if it exists")
-        context.drop_if_exists()
-        create_if_not_exists = True  # If schema is dropped, it needs to be recreated.
-
-    if create_if_not_exists:
-        logger.debug("Creating schema if not exists")
-        context.create_if_not_exists()
-
-    context.create_all()
+    try:
+        yaml_data = yaml.safe_load(file)
+        schema = Schema.model_validate(yaml_data)
+        url = make_url(engine_url)
+        if schema_name:
+            logger.info(f"Overriding schema name with: {schema_name}")
+            schema.name = schema_name
+        elif url.drivername == "sqlite":
+            logger.info("Overriding schema name for sqlite with: main")
+            schema.name = "main"
+        if not url.host and not url.drivername == "sqlite":
+            dry_run = True
+            logger.info("Forcing dry run for non-sqlite engine URL with no host")
+
+        metadata = MetaDataBuilder(schema).build()
+        logger.debug(f"Created metadata with schema name: {metadata.schema}")
+
+        engine: Engine | MockConnection
+        if not dry_run and not output_file:
+            engine = create_engine(url, echo=echo)
+        else:
+            if dry_run:
+                logger.info("Dry run will be executed")
+            engine = DatabaseContext.create_mock_engine(url, output_file)
+            if output_file:
+                logger.info("Writing SQL output to: " + output_file.name)
+
+        context = DatabaseContext(metadata, engine)
+
+        if drop and initialize:
+            raise ValueError("Cannot drop and initialize schema at the same time")
+
+        if drop:
+            logger.debug("Dropping schema if it exists")
+            context.drop()
+            initialize = True  # If schema is dropped, it needs to be recreated.
+
+        if initialize:
+            logger.debug("Creating schema if not exists")
+            context.initialize()
+
+        context.create_all()
+    except Exception as e:
+        logger.error(e)
+        raise click.ClickException(str(e))
```

Comment:
I forget, does click absorb the exception and convert this to an exit status so there is no need for `from e` here?

Reply:
That's what it seems to do, yes.